### DISCUSSION TITLE
MailboxAddress formating with folding

### DISCUSSION THREAD

**Discussion by jlynema** [jstedfast/MimeKit#1094] [Q&A] [answered]
I am using Mimekit to create and send SMTP messages and has been going well, but we are running into an unexpected problem with long friendly names in the "FROM" header.   
As an example, when I create a MailboxAddress to assign to the from with the following 
var fromAddress = new MailboxAddress("allowable length of some number of char is exactly the allowable length of some number of char.", "testaddressWith-Friendly@testing.address.ci");

I get the following output when I "ToString(false)"

```
"allowable length of some number of char is exactly the allowable length of some number of char." <testaddressWith-Friendly@testing.address.ci>
```

but the following output when I "ToString(true)"

```
allowable length of some number of char is exactly the allowable length of
 "some number of char." <testaddressWith-Friendly@testing.address.ci>
```

The resulting Mime when I serialize with WriteTo comes out as follows:

```
From: allowable length of some number of char is exactly the allowable length
 of "some number of char." <testaddressWith-Friendly@testing.address.ci>
```

Is this correct?  This is causing problems downstream for us with other mail clients.   I've tried setting FormatOptions.International as well, but that doesn't change the output.
When the friendly name is even longer it gets more interesting

```
From: allowable length of some number of char is exactly the allowable length
 of
 "some number of char. allowable length of some number of char is exactly the allowable length of some number of char."
	<testaddressWith-Friendly@testing.address.ci>
```

**Comment by jlynema** [comment]
For my code, it seems to fold and add extra quotes once the length of the friendly name hits 77 characters

**Reply by jstedfast** [reply, marked as answer]
These examples are all the expected result except maybe the last one which seems like the logic could be improved but is still valid output.

What you have to understand is that the rules for addresses make the following criteria:

1. `.` needs to be quoted
2. quoted-strings are meant to get *unquoted* by the parser on the receiving end
3. If the sending client wishes for the double-quotes to remain, then it must do `"This is a \"nested quote\" which will remain" <email@address.com>`

Once you understand all of those rules, you can clearly see that the output that MimeKit produces is correct and that the display of the receiving mail client (Outlook?) is wrong.

There is a similar discussion here: https://github.com/jstedfast/MimeKit/issues/1093
